addons/mail/tests/common.py

```diff
@@ -1462,10 +1463,18 @@ def _activate_multi_lang(cls, lang_code='es_ES', layout_arch_db=None, test_recor
             cls.env['ir.model']._get(test_record._name).with_context(lang=lang_code).name = 'Spanish Model Description'
 
         # Translate some code strings used in mailing
-        code_translations.python_translations[('test_mail', 'es_ES')]['NotificationButtonTitle'] = 'SpanishButtonTitle'
+        code_translations.python_translations[('test_mail', 'es_ES')] = {
+            **code_translations.python_translations[('test_mail', 'es_ES')],
+            'NotificationButtonTitle': 'SpanishButtonTitle'
+        }
         cls.addClassCleanup(code_translations.python_translations[('test_mail', 'es_ES')].pop, 'NotificationButtonTitle')
-        code_translations.python_translations[('mail', 'es_ES')]['View %s'] = 'SpanishView %s'
+        cls.addClassCleanup(ReadonlyDict, code_translations.python_translations[('test_mail', 'es_ES')])
```

Comment:
That doesn't do anything no ?

It will just call `ReadOnlyDict` on that argument but not replace the content of `code_translations.python_translations`.

```suggestion
        cls.addClassCleanup(lambda: code_translations.python_translations[('test_mail', 'es_ES')] = ReadonlyDict(code_translations.python_translations[('test_mail', 'es_ES')]))
```
?

To be honest it seems even over-kill for the actual need. Couldn't you just clear the cache as a cleanup ? And keep only this one cleanup ? It was filled within this test a few lines above anyway, with these `_update_translations` calls, so technically it's the test who filled this dict in the first place, so an actual accurate cleanup would be to empty what this test filled in the first place.

```suggestion
        cls.addClassCleanup(code_translations.python_translations.clear)
```